app/code/Magento/Ui/view/base/web/js/grid/url-filter-applier.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'uiComponent',
+    'underscore'
+], function (Component, _) {
+    'use strict';
+
+    return Component.extend({
+        defaults: {
+            filterProvider: 'componentType = filters',
+            filterKey: 'filters',
+            searchString: location.search,
+            modules: {
+                filterComponent: '${ $.filterProvider }'
+            }
+        },
+
+        /**
+         * Init component
+         *
+         * @return {exports}
+         */
+        initialize: function () {
+            this._super();
+            this.apply();
+
+            return this;
+        },
+
+        /**
+         * Apply filter
+         */
+        apply: function () {
+            var urlFilter = this.getFilterParam(this.searchString);
+
+            if (_.isUndefined(this.filterComponent())) {
+                setTimeout(function () {
+                    this.apply();
+                }.bind(this), 100);
+
+                return;
+            }
+            if (Object.keys(urlFilter).length) {
+                this.filterComponent().setData(urlFilter, false);
```

Comment:
How this component is behaving if the column/filter in the URL is not present or hidden on the grid?
Or in case there are URL parameters that are not related to the filters?

Reply:
- If the column/filter is not present it should behave the same way as the normal filter work, but I will test this scenario.
- If other parameters are added these are ignored. We are just looking for the `filters` parameter.